client/jetpack-connect/controller.js

```diff
@@ -147,6 +139,25 @@ export function connect( context, next ) {
 			persistMobileRedirect( '' );
 		}
 	}
+	next();
+}
+
+export function setMasterbar( context, next ) {
+	if ( config.isEnabled( 'jetpack/connect/mobile-app-flow' ) ) {
+		context.store.dispatch( setHideMasterbar( !! retrieveMobileRedirect() ) );
```

Comment:
referenced in a comment below, I find this naming confusing and it's evident here with `setHideMasterbar( boolean )` - what should that mean?

what if instead we had `setMasterbarVIsibility( boolean )` or even exposed two versions of the setter for additional clarity…

```js
const masterbarToggle = retrieveMobileRedirect() ? hideMasterbar() : showMasterbar();
context.store.dispatch( masterbarToggle );
```

Reply:
Yep, totally agree.